agent/provider/src/market/negotiator/builtin/expiration.rs

```diff
@@ -136,10 +145,10 @@ impl NegotiatorComponent for LimitExpiration {
                     // Below certain timeout it is impossible for Requestor to accept DebitNotes.
                     if req_deadline.num_seconds() < self.min_deadline {
                         return Ok(NegotiationResult::Reject {
-                            reason: Some(Reason::new(format!(
-                                "To low DebitNotes timeout: {}",
-                                req_deadline.display()
-                            ))),
+                            reason: Some(reason_with_extra(
+                                format!("To low DebitNotes timeout: {}", req_deadline.display()),
+                                serde_json::json!({"golem.proposal.rejection.is-final": true}),
```

Comment:
It would be much better, if `NegotiationResult` would have `isFinal` field. It would be responsibility of external layer to add this entry to Reason struct.
Note that this is field has special meaning for negotiation framework